src/Controller/ResourceController.php

```diff
@@ -149,104 +170,102 @@ private function getResponseFormat(array $validRdfFormats, Request $request)
+       
         $extension = array_pop($filenameChunks);
+
         $index = array_search($extension, array_column($validRdfFormats, 'extension'));
         if ($index !== false) {
             // This is a RDF file.
             $inputFormat = $validRdfFormats[$index]['format'];
 
             // Converting RDF from something to something else.
             $graph = new \EasyRdf_Graph();
-            $graph->parseFile($path, $inputFormat, $request->getUri());
+            $graph->parseFile($requested_path, $inputFormat, $request->getUri());
             $content = $graph->serialise($responseFormat);
-
+            $response->setContent($content);
             $headers = [
-                "Last-Modified" => $modifiedTime->format(\DateTime::W3c),
-                "Link" => [
-                    "<{$this->LDP_NS}Resource>; rel=\"type\"",
-                    "<{$this->LDP_NS}RDFSource>; rel=\"type\""
-                ],
-                "Vary" => "Accept",
-                "Content-Length" => strlen($content)
+                "Link" => ["<".self::LDP_NS."Resource>; rel=\"type\"",
+                           "<".self::LDP_NS."RDFSource>; rel=\"type\"" ],
+                "Content-Length" => strlen($content),
             ];
 
             $index = array_search($responseFormat, array_column($validRdfFormats, 'format'));
             if ($index !== false) {
                 $headers['Content-Type'] = $validRdfFormats[$index]['mimeType'];
             }
+            $response->headers->add($headers);
         } else {
             // This is not a RDF file.
-            $contentLength = filesize($path);
-            $responseMimeType = mime_content_type($path);
+            $contentLength = filesize($requested_path);
+            $responseMimeType = mime_content_type($requested_path);
+            $filename = explode("/", $requested_path);
+            $filename = end($filename);
             $headers = [
-                "Last-Modified" => $modifiedTime->format(\DateTime::W3C),
                 "Content-Type" => $responseMimeType,
-                "Link" => ["<{$this->LDP_NS}Resource>; rel=\"type\"",
-                           "<{$this->LDP_NS}NonRDFSource>; rel=\"type\""],
+                "Link" => ["<".self::LDP_NS."Resource>; rel=\"type\"",
+                           "<".self::LDP_NS."NonRDFSource>; rel=\"type\""],
                 "Content-Length" => $contentLength,
+                "Content-Disposition" => "attachment; filename=\"{$filename}\"",
             ];
 
-            if ($doGet) {
-                $stream = function () use ($path) {
-                    readfile($path);
+            $response->headers->add($headers);
+
+            if ($request->getMethod() == 'GET') {
+                $stream = function () use ($requested_path) {
+                    readfile($requested_path);
                 };
 
-                return $app->stream($stream, 200, $headers);
+                return $app->stream($stream, 200, $response->headers);
```

Comment:
I get this error: `Argument 3 passed to Silex\\Application::stream() must be of the type array, object given`